lib/nginx.js

```diff
@@ -0,0 +1,94 @@
+var nginxConf = require('./nginx-conf').genNginxConf;
+var configHash = require('./nginx-conf').getConfigHash;
+var path = require('path');
+var debug = require('debug')('strong-nginx-controller:nginx');
+var childProcess = require('child_process');
+var async = require('async');
+
+function Nginx(options) {
+  this._app = options.app;
+  this._nginxPath = options.nginxPath;
+  this._baseDir = options.baseDir;
+  this._nginxConf = nginxConf.bind(null, options);
+  this._configHash = configHash.bind(null, options);
+}
+
+/**
+ * Generate the nginx config and start nginx.
+ */
+function start(callback) {
+  async.series([
+    this._nginxConf,
+    this._cmd.bind(this, 'start')
+  ], callback);
+}
+Nginx.prototype.start = start;
+
+/**
+ * Generate a new nginx config. But only reload nginx if the config has changed.
+ */
+function reload(callback) {
+  var self = this;
+
+  self._configHash(function(err, originalHash) {
```

Comment:
The shape of this makes me think `async.waterfall` or `async.compose`.. but that's a really minor nit.